src/Microsoft.ML.Data/Transforms/Hashing.cs

```diff
@@ -472,6 +580,13 @@ public uint HashCore(uint seed, uint mask, in float value)
                 => float.IsNaN(value) ? 0 : (Hashing.MixHash(Hashing.MurmurRound(seed, FloatUtils.GetBits(value == 0 ? 0 : value))) & mask) + 1;
         }
 
+        private readonly struct HashFloatV2 : IHasher<float>
+        {
+            [MethodImpl(MethodImplOptions.AggressiveInlining)]
+            public uint HashCore(uint seed, uint mask, in float value)
```

Comment:
>HashCore [](start = 24, length = 8)

The `1` still has to be added for all the new `V2` implementations. It is added because the output of `HashingTransformer` is a key type, and 0 is the `NA` value of key types.

Reply:
gotcha, but what if the hash value is uint.max? Wouldn't that end up mapping to 0?

Reply:
This is why we only allow hashing into at most 31 bits, to prevent this situation.

---
In reply to: [416030419](https://github.com/dotnet/machinelearning/pull/5013#discussion_r416030419) [](ancestors = 416030419)